### PR TITLE
new splitting function

### PR DESCRIPTION
**Pull request by mariamanoli**
This is a pull request for the new energy splitting model DAZLS. The current branch contains two new files that need to be reviewed: better_dazls.py and better_dazls.ipynb

### CODE REVIEW

File: openstef/tasks/split_forecast.py [hunk 1]
```diff
@@ -69,7 +72,7 @@ def split_forecast_task(
     id pid.
 
     Args:
-        pid (int): Prediction job id
+        pj (int): Prediction job id
```
Comment (FrankKr) [inline openstef/tasks/split_forecast.py:75]: What is used here; the id of the prediction_job, or the complete prediction_job?
If only the id is used, please call the variable `pid`, if the whole prediction_job is used, please change the `(int)` to `PredictionJob`

File: openstef/model/regressors/better_dazls.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            "sem1",
+        ]
+        self.target_columns_index = ["total_wind_part", "total_solar_part"]
+
+    def fit(self, features, target):
+        """
+        In this function we scale the input of the domain and adaptation models of the DAZLS MODEL.
+        Then we fit the two models.
+        With the help of the index we separate the features into domain_model_input, adaptation_model_input and the
+        target, and we use them for the fitting and the training of the models.
```
Comment (FrankKr) [inline openstef/model/regressors/better_dazls.py:63]: Nice docstring!

File: openstef/model/regressors/better_dazls.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+from sklearn.base import BaseEstimator
+
+
+# DAZLS algorithm
+class BetterDazls(BaseEstimator):
+    """
+    The model carries out wind and solar power prediction for unseen target substations using training data from other substations with
+    known components.
+
+    Any data-driven model can be plugged and used as the base for the domain and the adaptation model.
```
Comment (FrankKr) [inline openstef/model/regressors/better_dazls.py:15]: Nice docstring. Is there already a link available with more elaborate documentation on the model?

**Review by FrankKr — CHANGES_REQUESTED** [review]
Nice job combining all the stuff!
Suggestions: 
- Move the better_dazls.ipynb to icarus-analyses
- Move all the csv files to icarus-analyses
- Run to notebook to train a model on your laptop, store that model, and add the stored model to somewhere in openstef

File: openstef/model/regressors/better_dazls.py (hunk 3 above)
Comment (mariamanoli) [reply openstef/model/regressors/better_dazls.py:15]: The scientific paper has more information about the model but it is not publicly available yet.

File: openstef/model/regressors/better_dazls.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
+# SPDX-FileCopyrightText: 2017-2022 Contributors to the OpenSTEF project <korte.termijn.prognoses@alliander.com> # noqa E501>
```
Comment (JanMaartenvanDoorn) [inline openstef/model/regressors/better_dazls.py:1]: Also good to rename better_dazls to just dazls

**Review by JanMaartenvanDoorn — COMMENTED** [review]
Nice addition to the code! Added some remarks to improve clarity
In addition i is also good to remove all the code related to the old splitting method. We do not need the code to determine the splitting coefficients anymore.

File: openstef/pipeline/create_component_forecast.py [hunk 5]
```diff
@@ -2,26 +2,71 @@
...
+    :param input_data: (pd.DataFrame): Input forecast for the components forecast.
+    :param weather_data: (pd.DataFrame): Weather data with 'radiation' and 'windspeed_100m' columns
+
+    :return: input_df (pd.Dataframe): It outputs a dataframe which will be used for the Dazls prediction function.
+    """
+
+    # Prepare raw input data
+    input_df = weather_data[["radiation", "windspeed_100m"]].merge(
+        input_data[["forecast"]].rename(columns={"forecast": "total_substation"}),
+        how="inner",
```
Comment (MartijnCa) [inline openstef/pipeline/create_component_forecast.py:33]: Your code looks quite readable to me, eventhough I am new to the repository. Good job on that!

Considering this merge I was wondering why you chose to do an "inner"-merge, as opposed to a right merge for example? Does there ever occur a situation in which you would want rows from the right dataframe to be excluded after the merge?

**Review by FrankKr — APPROVED** [review]
Looks really clear!
I made some minor remarks on naming conventions.

**Review by MartijnCa — APPROVED** [review]
Looks good Maria! I only posted one comment for a little bit of clarification.

**Pull request merged**